index.js

```diff
@@ -119,22 +119,25 @@ function resolveLocalRelativeImport(fileImport, file) {
  * @param {String} rootPath: root path of the project code
  */
 function resolveWebpackShim(source, kibanaPath, rootPath) {
-  const pluginShimPath = path.join(rootPath, 'webpackShims');
-  const pluginMatches = getFileMatches(source, pluginShimPath);
-  const pluginFileMatches = getMatch(pluginMatches, pluginShimPath);
-  debug(`resolveWebpackShim: checking for ${source}`);
-  if (pluginFileMatches.found) {
-    debug(`resolved webpackShim import in plugin: ${source}`);
-    return pluginFileMatches;
-  }
+  debug(`resolveWebpackShim: resolving ${source}`);
+  const sourceParts = source.split('/');
+  const baseSource = sourceParts.pop();
+
+  const pluginShimPaths = [
+    path.join(rootPath, 'webpackShims', ...sourceParts),
```

Comment:
The `webpackShims` directory could be in any directory relative to the file up to the root, so this needs to check for `webpackShims` directories starting at the basedir of the requested file up to the root. The console plugin, for instance, has it's own webpackShims directory, and any directory can have a webpackShims directory just like it can have a node_modules directory.